contracts/src/v0.8/vrf/VRFV2PlusWrapperConsumerBase.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/LinkTokenInterface.sol";
+import "../interfaces/VRFV2PlusWrapperInterface.sol";
+
+/** *******************************************************************************
+ * @notice Interface for contracts using VRF randomness through the VRF V2 wrapper
+ * ********************************************************************************
+ * @dev PURPOSE
+ *
+ * @dev Create VRF V2 requests without the need for subscription management. Rather than creating
+ * @dev and funding a VRF V2 subscription, a user can use this wrapper to create one off requests,
+ * @dev paying up front rather than at fulfillment.
+ *
+ * @dev Since the price is determined using the gas price of the request transaction rather than
+ * @dev the fulfillment transaction, the wrapper charges an additional premium on callback gas
+ * @dev usage, in addition to some extra overhead costs associated with the VRFV2Wrapper contract.
+ * *****************************************************************************
+ * @dev USAGE
+ *
+ * @dev Calling contracts must inherit from VRFV2WrapperConsumerBase. The consumer must be funded
+ * @dev with enough LINK to make the request, otherwise requests will revert. To request randomness,
+ * @dev call the 'requestRandomness' function with the desired VRF parameters. This function handles
+ * @dev paying for the request based on the current pricing.
+ *
+ * @dev Consumers must implement the fullfillRandomWords function, which will be called during
+ * @dev fulfillment with the randomness result.
+ */
+abstract contract VRFV2PlusWrapperConsumerBase {
+  LinkTokenInterface internal immutable LINK;
+  VRFV2PlusWrapperInterface internal immutable VRF_V2_WRAPPER;
```

Comment:
I agree that for most cases, wrapper itself will migrate to a new v2plus coordinator but I think it doesn't hurt to make this mutable, in case new wrapper does need to be deployed.

Reply:
Good point @jinhoonbang I think making this mutable is reasonable. We have to commit to keeping the `transferAndCall` and native call interface the same across V2+ wrappers though for the cutover to a new wrapper to be seamless.